agent/run.sh

```diff
@@ -235,10 +235,17 @@ upgrade()
 
 wait_for()
 {
+    info "Atempting to connect to: ${CATTLE_URL}"
     for ((i=0; i < 300; i++)); do
         if ! curl -f -s ${CATTLE_URL} >/dev/null 2>&1; then
             error "${CATTLE_URL}" is not accessible
+            sleep 2
+            if [ "$i" -eq "299" ]; then
+                error "Could not reach ${CATTLE_URL}. Giving up."
+                exit 1
+            fi
         else
+            info "${CATTLE_URL} is accessible"
```

Comment:
Why doing something else it dawned on me that echoing CATTLE_URL will cause a problem for support.  When authentication is turned on the CATTLE_URL has a registration token embedded in it which is private to that account.  We don't want to have that URL in logs.  You could sanitize the CATTLE_URL.  So the registration token will look like `http://localhost:8080/v1/scripts/81AB22FBCF352E4498E2:1422684000000:HuMjQsZJPxiBsGYpFo0FrNVv2I`.  So basically `sed 's!/v1/scripts.*!/v1!`